### PR TITLE
Add Grunt support and use WebRTC Adapter.js from Bower dependency

### PR DESCRIPTION
**Pull request by hthetiot**
Add Grunt support and use WebRTC Adapter.js from Bower dependency.

Detailed features:
- UMD support for easyrtc
- build via gunt+ requirejs (uglify later)
- lint (css/html/js) via grunt+csslint+jshint htmlhint
- basic unit test env for api via grunt+karma+jasmine
- generated doc via grunt+jsdoc
- deprecated message on previous build scripts

New main grunt commands:
> $ grunt build 
> $ grunt build_api
> $ grunt jsdoc

Used Adapter.js via bower https://github.com/webrtc/adapter

### CODE REVIEW

**Comment by hthetiot** [comment]
Please review.

cc @mike-anderson @omochi

**Comment by omochi** [comment]
Sorry I don't touch webrtc development recently.

**Comment by mike-anderson** [comment]
Sorry, I'm not an active contributor to easyrtc anymore.
My only comment would be that it looks to me like the refactor moved a large amount of file content and git was not able to properly track it. I would see if there is a way to inform in the commit what has moved, to preserve more of the code history.

File: api/easyrtc_int.js [hunk 1]
```diff
@@ -1,46 +1,73 @@
-/** @class
- *@version 1.0.16-beta
- *<p>
- * Provides client side support for the EasyRTC framework.
- * Please see the easyrtc_client_api.md and easyrtc_client_tutorial.md
- * for more details.</p>
+/*!
+  Script: easyrtc.js
```
Comment (hthetiot) [inline api/easyrtc_int.js]: Cleanup header.

File: api/easyrtc_int.js [hunk 2]
```diff
@@ -1,46 +1,73 @@
...
+    SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+    CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+
+/* global MediaStreamTrack, createIceServer, RTCIceCandidate, RTCPeerConnection, RTCSessionDescription */ // WebRTC
+
+(function (root, factory) {
```
Comment (hthetiot) [inline api/easyrtc_int.js]: Add UMD to script.

File: api/easyrtc_int.js [hunk 3]
```diff
@@ -1,46 +1,73 @@
...
+ * @constructs Easyrtc
  */
-
-/* global getUserMedia, MediaStreamTrack, createIceServer, RTCIceCandidate, RTCPeerConnection, RTCSessionDescription */ // WebRTC
-/* global webrtcDetectedBrowser, webrtcDetectedVersion, attachMediaStream , webrtcUtils*/ // adapter.js
-/* global easyrtc_constantStrings */ // easyrtc_lang_en.js
-/* global io */
-
 var Easyrtc = function() {
+
```
Comment (hthetiot) [inline api/easyrtc_int.js]: Factorize log calls

File: api/easyrtc_int.js [hunk 4]
```diff
@@ -204,7 +229,7 @@ var Easyrtc = function() {
 
     /** @private */
     //
-    // Maps a key to a language specific string using the easyrtc_constantStrings map.
```
Comment (hthetiot) [inline api/easyrtc_int.js]: rename easyrtc_constantStrings to easyrtc_lang

File: api/easyrtc_int.js [hunk 5]
```diff
@@ -398,7 +423,7 @@ var Easyrtc = function() {
      * @param value - true to receive video, false otherwise. The default is true.
      */
     this.enableVideoReceive = function(value) {
-        if (webrtcDetectedBrowser === "firefox") {
+        if (adapter && adapter.browserDetails && adapter.browserDetails.browser === "firefox") {
```
Comment (hthetiot) [inline api/easyrtc_int.js]: Use new Adapter

**Comment by dcalacci** [comment]
@hthetiot hey, nice work. How is this work going? It'd be really nice to be able to use the easyRTC client with npm / browserify / webpack...

**Comment by hthetiot** [comment]
@dcalacci it is stable I'm using the grunt branch in prod already.

Here is the new stuff: UMD support, new build via grunt, use official webrtc adapter from bower, and some lint and regression, MS edge to MS Edge support.

I will merge in beta once @EricDavies and I released beta to master.

File: api/adapter.js [hunk 6]
```diff
@@ -1,1711 +0,0 @@
-/*
```
Comment (hthetiot) [inline api/adapter.js]: Removed and using "webrtc-adapter" from bower.json now instead.

**Pull request merged**